From d16e067723899aa9023e9c1070ffcf52e6ec2544 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:29 -0400
Subject: [ethernet] igb: Don't let ethtool try to write to iNVM in i210/i211

Message-id: <1409062359-13063-12-git-send-email-sassmann@redhat.com>
Patchwork-id: 89130
O-Subject: [RHEL7.1 PATCH 11/81] igb: Don't let ethtool try to write to iNVM in i210/i211
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Fujinaka, Todd" <todd.fujinaka@intel.com>

Don't let ethtool try to write to iNVM in i210/i211.

This fixes an issue seen by Marek Vasut.

Reported-by: Marek Vasut <marex@denx.de>
Signed-off-by: Todd Fujinaka <todd.fujinaka@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a71fc313c4f569be5788caff07ef1fe346842c5b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index 0ae3177..b918ba3 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -771,8 +771,10 @@ static int igb_set_eeprom(struct net_device *netdev,
  if (eeprom->len == 0)
   return -EOPNOTSUPP;
 
- if (hw->mac.type == e1000_i211)
+ if ((hw->mac.type >= e1000_i210) &&
+     !igb_get_flash_presence_i210(hw)) {
   return -EOPNOTSUPP;
+ }
 
  if (eeprom->magic != (hw->vendor_id | (hw->device_id << 16)))
   return -EFAULT;
-- 
1.7.1